### PR TITLE
update to drop Julia pre-1.0 code, add v1.1 upgrades

### PR DESCRIPTION
**Pull request by vtjnash**
Add some tooling for converting build paths to installation paths for Julia's base (and stdlib) source files.

Julia v1.1 is slated to have a new, more flexible code-coverage format (https://github.com/JuliaLang/julia/pull/30381). This is the prep-work (e.g. helper functions) for utilizing that information.

### CODE REVIEW

File: Manifest.toml [hunk 1]
```diff
@@ -1,146 +0,0 @@
...
-[[Test]]
-deps = ["Distributed", "InteractiveUtils", "Logging", "Random"]
-uuid = "8dfed614-e22c-5e08-85e1-65c5234f0b40"
-
-[[UUIDs]]
-deps = ["Random", "SHA"]
-uuid = "cf7118a7-6976-5b1a-9a39-7adc72f591a4"
-
-[[Unicode]]
-uuid = "4ec0a83e-493e-50e2-b9ac-8f72acf5a8f5"
```
Comment (staticfloat) [inline Manifest.toml]: I think it is useful to keep a `Manifest.toml` file around as a record of one possible set of packages that is known to work.  It's not used during resolution, but it at least proves that on your machine, things worked, and can be updated as necessary.

File: src/CoverageBase.jl [hunk 2]
```diff
@@ -1,33 +1,121 @@
...
-    _spawn(cmd) = run(pipeline(cmd, stdin=devnull), wait=false)
-else
-    _spawn(cmd) = spawn(cmd, devnull, stdout, stderr)
+function __init__()
+    global _julia_top, julia_datapath
+    dir = String(normpath(Sys.BINDIR::String, Base.DATAROOTDIR, "julia", ""))
+    julia_datapath = dir
+    if !(isdir(joinpath(dir, "base")) && isdir(joinpath(dir, "test")))
+        # this branch shouldn't happen normally
+        dir = Sys.BINDIR::String
```
Comment (staticfloat) [inline src/CoverageBase.jl:15]: Out of curiosity, why the `::String` type assertion?

**Review by staticfloat — APPROVED** [review]

File: src/CoverageBase.jl (hunk 2 above)
Comment (vtjnash) [reply src/CoverageBase.jl:15]: habit—it makes the `__init__` method inferrable, and thus makes the package probably faster to load

File: Manifest.toml (hunk 1 above)
Comment (vtjnash) [reply Manifest.toml]: Sure, sounds logical. I wanted to leave comments in the Project.toml file also to explain how some of these versions and dependencies happen, but it seemed like Pkg3 doesn't actually support that.

**Pull request merged**